test/unit/effects.js

```diff
@@ -1028,7 +1028,7 @@ jQuery.each( {
 				jQuery( elem ).remove();
 
 			} );
-			this.clock.tick( 50 );
+			this.clock.tick( 100 );
```

Comment:
Funny you should ask... the test uses [`jQuery.when( elem ).done(…)`](https://github.com/gibson042/jquery/blob/gh-3029-jQuery-when-deferred/test/unit/effects.js#L963), and this PR updates the single-argument form of `jQuery.when` to [use `.then()`](#diff-00843d85b3ae02d8ca45835b8d0a4edfR371) for unwrapping secondary thenables (e.g., `jQuery.when( jQuery.Deferred().resolve( thenable ) )`). Single-argument `jQuery.when` becomes necessarily asynchronous as a side effect, and we therefore require an extra clock tick to invoke the callback.